From e02bbc5ebda1ab8cf8616829a03f96a4e4918e03 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 17 Nov 2013 14:05:06 +0000
Subject: [PATCH] MythMusic: fix compilation for users of taglib < 1.8

Seems you need tablib >= 1.8 to remove Picture blocks from xiph comments
but the minimum we currently support is 1.6 so since this is a rarely used
feature just disable it on taglib < 1.8 for now.

Refs #11947.
---
 mythtv/libs/libmythmetadata/metaioflacvorbis.cpp |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp b/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp
index 16c9ff9..22b6809 100644
--- a/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp
+++ b/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp
@@ -196,7 +196,7 @@ TagLib::FLAC::Picture *MetaIOFLACVorbis::getPictureFromFile(
         const TagLib::List<Picture *>& picList = flacfile->pictureList();
 
         for (TagLib::List<Picture *>::ConstIterator it = picList.begin();
-             it != picList.end(); it++)
+                it != picList.end(); it++)
         {
             pic = *it;
             if (pic->type() == artType)
@@ -335,6 +335,7 @@ AlbumArtList MetaIOFLACVorbis::getAlbumArtList(const QString &filename)
 bool MetaIOFLACVorbis::writeAlbumArt(const QString &filename,
                               const AlbumArtImage *albumart)
 {
+#if TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION >= 8
     using TagLib::FLAC::Picture;
     if (filename.isEmpty() || !albumart)
         return false;
@@ -390,6 +391,11 @@ bool MetaIOFLACVorbis::writeAlbumArt(const QString &filename,
     }
 
     return retval;
+#else
+    LOG(VB_GENERAL, LOG_WARNING,
+        "TagLib 1.8.0 or later is required to write albumart to flac xiphComment tags");
+    return false;
+#endif
 }
 
 /*!
@@ -402,6 +408,8 @@ bool MetaIOFLACVorbis::writeAlbumArt(const QString &filename,
 bool MetaIOFLACVorbis::removeAlbumArt(const QString &filename,
                                const AlbumArtImage *albumart)
 {
+#if TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION >= 8
+
     if (filename.isEmpty() || !albumart)
         return false;
 
@@ -435,6 +443,11 @@ bool MetaIOFLACVorbis::removeAlbumArt(const QString &filename,
     }
 
     return retval;
+#else
+    LOG(VB_GENERAL, LOG_WARNING,
+        "TagLib 1.8.0 or later is required to remove albumart from flac xiphComment tags");
+    return false;
+#endif
 }
 
 bool MetaIOFLACVorbis::changeImageType(const QString &filename,
-- 
1.7.10.2

